cornell-dti/course-plan
# Self-check and related modal fixes

### Summary 

Fix a number of issues with self-check and related modals.

- [x] Filter out courses from the self-check dropdown if they do not pass that requirements checker (part 2 of the larger self-check fix)
- [x] Convert edit & delete sem to flexible modal (can now be clicked outside)
- [x] Have dropdowns reset properly for the new self check add modal and edit sem modal
- [x] Have all new semester dropdowns (so new semester, edit semester, and new self check add modal) have gray placeholder text until selected. 

![image](https://user-images.githubusercontent.com/25535093/112728709-edf49a80-8efe-11eb-9e04-e3dd680e01d0.png)

### Test Plan 

1. Confirm that courses that do not fulfill self-check reqs with warnings do not show up in the dropdown, but do otherwise. An example would be "INFO 1998" when added to the plan should not show up in the CS External/Technical reqs, but should show up in the Engineering Advisor Approved Self-Check. A course like INFO 4300 should still show up in external/technical reqs.
2. Confirm edit semester and delete semester look and work as before, but can now be clicked outside
3. Confirm edit semester dropdowns are reset after choosing a season or year, and then clicking outside the modal, and clicking either the 'x' or 'cancel' button.
4. Confirm the same for the new self-check add modal
5. Confirm that edit semester, new semester, and self-check add modals all have gray text when opening now until the year/season is changed. Confirm they reset to gray when opening again.
6. In general, confirm courses can still be added from the self-check modal, reqs can still be assigned from self-check dropdown, and sems can still be added/edited/deleted
